k8s/base/deploy-frontend/deploy.yaml

```diff
@@ -0,0 +1,79 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/master-standalone/deployment-apps-v1.json
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: frontend
+  labels:
+    app.kubernetes.io/component: frontend
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: frontend
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: frontend
+    spec:
+      securityContext:
+        fsGroup: 1000
+      volumes:
+        - name: dist
+          emptyDir:
+            sizeLimit: 256Mi
+        - name: app-config
+          configMap:
+            name: app-config
+      initContainers:
+        - name: build-files
+          image: lco-education-platform
+          command:
+            - "sh"
+            - "-c"
+            - "cp -r /app/dist/* /app/volume"
+          securityContext:
+            runAsNonRoot: true
+            runAsUser: 1000
+            runAsGroup: 1000
+            readOnlyRootFilesystem: true
+          envFrom:
+            - secretRef:
+                name: env
+                optional: false
+          resources:
+            requests:
+              cpu: 1000m
+              memory: 256Mi
+            limits:
+              cpu: 2000m
+              memory: 512Mi
+          volumeMounts:
+            - name: dist
+              mountPath: /app/volume
+      containers:
+        - name: default
+          image: ghcr.io/static-web-server/static-web-server:2
+          command:
+            - /static-web-server
+            - --root=/app/dist
+            - --health
+            - --port=8080
+            - --host=0.0.0.0
+            - --page-fallback=/app/dist/index.html
```

Comment:
This `--page-fallback` is necessary to force the static server to rely on the vue router (by routing to index.html) to route the browser. See https://static-web-server.net/features/error-pages/ for more details. Without it, when navigating to anything other than the root page, the server assumes it should serve `/app/dist + /url/path` which results in a 404, since the views aren't organized exactly as the paths are.

Reply:
this is really good to know!